packages/components/ng-log-to-customer/src/controller.js

```diff
@@ -34,17 +34,37 @@ export default class LogToCustomerCtrl {
     this.keys = Object.keys(this.logKeys);
     this.setUrlTimeout = null;
 
-    this.setUrlSource().then(() => !this.errorMessage && this.startLog());
+    this.kinds = [];
+    this.LogToCustomer.getLogsKind(this.kindEndpoint)
+      .then((data) => {
+        this.kinds = data;
+        this.selectedKind = data[0]?.name;
+      })
+      .then(() => this.generateTmpUrlSource())
+      .then(() => !this.errorMessage && this.startLog())
+      .catch(this.setErrorMessage());
```

Comment:
declare a function here, dot not execute it directly.
By the way, not sure your catch is useful here are all called methods already catch errors

```suggestion
      .catch(this.setErrorMessage);
```

Reply:
The first service call doesn't have a catch that's why I put a catch here too